R/paginate.R

```diff
@@ -87,16 +102,19 @@ paginate_req_perform <- function(req,
   check_bool(progress)
 
   resp <- req_perform(req)
-  f_n_pages <- req$policies$paginate$n_pages %||% function(resp) Inf
-  n_pages <- min(f_n_pages(resp), max_pages)
+  parsed <- paginate_parse_response(resp, req)
+
+  f_n_pages <- req$policies$paginate$n_pages %||% function(resp, parsed) Inf
```

Comment:
Do this in `req_paginate()` so we're setting all the default functions in one place?